app/src/main/java/com/fbada/projectone/domain/Car.kt

```diff
@@ -0,0 +1,572 @@
+                in 700..999 -> HIGH
+                else -> ULTIMATE
+            }
+        }
+
+        /**
+         * Calculates a price score based on vehicle max speed.
+         */
+        private fun sprinterPriceScoreBasedOnMaxSpeed(): Int {
+            return when (maxSpeed) {
+                in 0..60 -> LOW
+                in 61..80 -> MEDIUM
+                in 81..100 -> HIGH
+                else -> ULTIMATE
+            }
+        }
+    }
+
+    /**
+     * Subclass Traviliner Model type car.  Extends [Mercedes] class.
+     *  @property powerSteering whether power steering available
+     */
+    class Traveliner(
+        var powerSteering: Boolean,
+        doors: Int,
+        maxSpeed: Int,
+        year: Int,
+        colour: AvailableColours,
+        fuelType: AvailableFuelTypes,
+        gearbox: AvailableGearBoxes
+    ) :
+        Mercedes(doors, maxSpeed, year, colour, fuelType, gearbox) {
+        /**
+         * @property price price based on max speed price score.
+         */
+        override var price: Double = super.price + (travelingPriceBasedOnMaxSpeed() * 3)
+
+        /**
+         * Calculates a score based on max speed.
+         */
+        private fun travelingPriceBasedOnMaxSpeed(): Int {
+            return when (maxSpeed) {
+                in 0..60 -> LOW
+                in 61..80 -> MEDIUM
+                in 81..100 -> HIGH
+                else -> ULTIMATE
+            }
+        }
+    }
+
+    /**
+     * Subclass Cougar Model type car.  Extends [Ford] class.
+     * @property radioType Type of radio inside car.
+     */
+    abstract class Cougar(
+        private val radioType: AvailableRadios,
+        seats: Int,
+        interior: AvailableInteriorMaterials,
+        owners: Int,
+        year: Int,
+        availableColours: AvailableColours,
+        fuelType: AvailableFuelTypes,
+        gearbox: AvailableGearBoxes
+    ) : Ford(seats, interior, owners, year, availableColours, fuelType, gearbox) {
+        /**
+         * @property price Base price plus interior score
+         */
+        override var price: Double = super.price + interiorPremium() * 15
+
+        /**
+         * Calculates a price based on interior materials and radio.
+         */
+        private fun interiorPremium(): Double {
+            val materialPriceScore = when (interiorMaterial) {
+                AvailableInteriorMaterials.LEATHER, AvailableInteriorMaterials.SUEDE -> HIGH
+                AvailableInteriorMaterials.COMBO -> HIGH
+                AvailableInteriorMaterials.FABRIC -> MEDIUM
+                else -> {
+                    LOW
+                }
+            }
+            val radioPriceScore = when (radioType) {
+                AvailableRadios.SMART_PHONE_CONNECT -> ULTIMATE
+                AvailableRadios.INTERNET, AvailableRadios.DIGITAL -> HIGH
+                AvailableRadios.CD_PLAYER -> LOW
+                else -> {
+                    MIN
+                }
```

Comment:
Feels like a good candidate for extracting into functions.

Reply:
Extracted into functions.